CONTRIBUTING.md

```diff
@@ -0,0 +1,91 @@
+
+If you’re planning on contributing a new feature or an otherwise complex contribution, we kindly ask you to start a conversation with the maintainer team by opening up a GitHub issue first.
+
+### General Guidelines
+
+A primary goal of the agent is to adhere to the Hippocratic oath for the applications it instruments: “first, do no harm”.  The code in the agent will execute inside the process space of our customers’ applications and must not crash them, destabilize them, change their behavior, or significantly degrade their performance.  Another important consideration is that the data the agent gathers must not leak any personally identifying information (PII) from customer apps, e.g. usernames or credit card numbers.
+
+This project is licensed under the Apache-2.0 license.  Any third party libraries added as dependencies of the project must have a similarly permissive open source license, e.g. MIT.
+
+### Coding Style Guidelines
+
+Our repository includes an [.editorconfig](https://github.com/newrelic/newrelic-dotnet-agent/blob/main/.editorconfig) which will be used automatically by Visual Studio to maintain consistent code formatting.
+
+#### Variable naming conventions
+
+| Format      | Use For | Example |
+| ----------- | ------- | ------- |
+| PascalCase  | Constant fields, public class level fields | `private const string MyConst = "NewRelic";` |
+| camelCase  | Local variables (note: we prefer the use of the `var` implicit definition keyword whenever possible) | `var myId = 12345;` |
+| _camelcase  | Private class-level fields, including static and/or read-only | `private int _myId;` |
+
+#### Class naming conventions
+
+- All class names should be written in PascalCase and should be singular, e.g. `TransactionName`.  If it is a collection, it should be pluralized e.g. `TransactionNames`.  Code files can contain multiple classes if the classes are tightly coupled, but this should be rare.
+- All class declarations should have access modifiers.
+
+#### Interface naming conventions
+
+- All interfaces names should be written in PascalCase and prefixed with the letter "I", e.g. `ITransactionName`.
+
+#### Method naming conventions
+
+- Methods should be named using PascalCase and parameters should be named using camelCase, e.g `private string GetUserNameFromId(int userId)`.
+
+#### Class file layout
+
+- The preferred order of declarations within a class is:
+  - Fields
+  - Properties
+  - Methods (with constructors grouped at the beginning)
+  - Events
+
+### Testing Guidelines
+
+See our [development](/docs/development.md) and [integration testing](/docs/integration-tests.md) documentation to run tests, including required setup steps.
```

Comment:
these link are broken until we merge https://github.com/newrelic/newrelic-dotnet-agent/pull/68
not a big deal, but we are holding that PR until the dirs match after the home builder work

Reply:
#68 has been merged now so these should be fine.